docs/devel/owners.md

```diff
@@ -0,0 +1,72 @@
+# Owners files
+
+_Note_: This is a design for a feature that is not yet implemented.
+
+## Overview
+We want to establish owners for different parts of the code in the Kubernetes codebase.  These owners
+will serve as the approvers for code to be submitted to these parts of the repository.  Notable, owners
+are not necessarily expected to do the first code review for all commits to these areas, but they are
+required to approve changes before they can be merged.
+
+## High Level flow
+
+### Step One: A PR is submitted
+After a PR is submitted, a user can request the approval from the required owners.  The user requests this
+by writing:
+```
+@k8s-bot approvers
+```
+
+into a comment.  When our automation sees this comment, it will reply with an annotation that indicates
+the owners required to approve.  The annotation is a comment that is applied to the PR.
+This comment will say:
+
+```
+Approval is required from <owner-a>, <owner-b>
```

Comment:
Oh, looks like you meant "and" (later you say "all owners have to approve") -- is this what we want? What if an owner is on an extended vacation? I guess we could say an owner should remove themselves from the owners file if they are going to be on an extended vacation. But then (1) the annotation for any existing PRs will be wrong (it will mention them when they are not actually required) and (2) once they return the annotation for PRs submitted while they were out will be wrong (it will not mention them but they are actually required). Of course I'm assuming the mergebot uses the current owners file, not whatever was recorded in the PR thread (this seems like the only reasonable policy). 

The easiest solution is to make it an "or" -- though I can see downsides to that too. But those at least could be solved by social agreement (for example, if A and B are owners, they could implicitly implement "and" as follows: they agree that normally neither will write "I approve" until both have "LGTMd" - the second to "LGTM" will write "I approve". But if one is on vacation, then the other could write "I approve" without an LGTM from the one on vacation.)

So I would suggest "or." I'm assuming we will have a small enough group of owners for each part of the codebase (e.g. 2-3 people) that such social agreements will be workable.